scipy/signal/_peak_finding_utils.pyx

```diff
@@ -148,3 +148,98 @@ def _select_by_peak_distance(np.intp_t[:] peaks not None,
                 k += 1
 
     return keep.base.view(dtype=np.bool)  # Return as boolean array
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def _peak_prominences(np.float64_t[::1] x not None,
```

Comment:
You can use a `@cython.nonecheck(False)` directive here instead of manually specifying `not None`.

See https://github.com/cython/cython/wiki/enhancements-compilerdirectives

Reply:
I think you mean `@cython.nonecheck(True)`. Using `False` would turn the check off which is off by defaut anyway. 

Using that directive is a good idea for #8594 which features this statement many times. But in this case I like the explicit statements more  as only two arguments use them. Do you agree?

Reply:
Ah, I must have mis-read that directive's docs. Current usage looks good.